Cubical/Homotopy/Loopspace.agda

```diff
@@ -22,3 +24,81 @@ open import Cubical.Foundations.GroupoidLaws
 {- loop space map -}
 Ω→ : ∀ {ℓA ℓB} {A : Pointed ℓA} {B : Pointed ℓB} (f : A →∙ B) → (Ω A →∙ Ω B)
 Ω→ (f , f∙) = (λ p → (sym f∙ ∙ cong f p) ∙ f∙) , cong (λ q → q ∙ f∙) (sym (rUnit (sym f∙))) ∙ lCancel f∙
+
+
+-- Eckmann-Hilton --
+
+{- Horizontal composition -}
+_⋆_ : ∀ {ℓ} {A : Type ℓ} {a b c : A} {p q : a ≡ b} {p' q' : b ≡ c} → (p ≡ q) → (p' ≡ q')
+   → p ∙ p' ≡ q ∙ q'
+_⋆_ α β i = α i ∙ β i
+
+{- Exchange law-}
+exchange : ∀ {ℓ} {A : Type ℓ} {a b c : A} {p q r : a ≡ b} {p' q' r' : b ≡ c}
+           (α : p ≡ q) (β : q ≡ r) (α' : p' ≡ q') (β' : q' ≡ r')
+        → (α ∙ β) ⋆ (α' ∙ β') ≡ (α ⋆ α') ∙ (β ⋆ β')
+exchange {A = A} {a = a} {c = c} {p = p} {r = r} {p' = p'} {r' = r'} α β α' β' z i =
+  hcomp (λ k → λ { (i = i0) → p ∙ p'
+                  ; (i = i1) → (β ⋆ β') (k ∨ ~ z)
+                  ; (z = i0) → ((α ∙ β) ⋆ (α' ∙ β')) i })
+        (bottom z i)
+  where
+  bottom : PathP (λ i → p ∙ p' ≡ (β ⋆ β') (~ i)) ((α ∙ β) ⋆ (α' ∙ β')) (α ⋆ α')
+  bottom i =
+    hcomp (λ k → λ { (i = i0) → (α ∙ β) ⋆ (α' ∙ β')
+                    ; (i = i1) → rUnit α (~ k) ⋆ rUnit α' (~ k)})
+          ((α ∙ λ j → β (~ i ∧ j)) ⋆ (α' ∙ λ j → β' (~ i ∧ j)))
+
+Eckmann-Hilton : ∀ {ℓ} {A : Pointed ℓ} (n : ℕ) (α β : typ ((Ω^ (2 + n)) A))
```

Comment:
Isn't there a more direct proof of this using J?

(pint @bbentzen)

Reply:
I never thought I'd hear you say that;-) I followed the proof in the HoTT book, but if there is a simpler proof I'd be happy to change it.

Reply:
If there's one thing I learned last year is that there's no shame in using J when necessary! @aljungstrom Maybe you can have a look at Theorem 6.2.1 [here](https://arxiv.org/pdf/1911.05844.pdf). The proof is given informally, so I don't know how easy it would be to mechanize it, but the general idea could be helpful!

Reply:
@bbentzen Sorry for the late reply. Thanks for the tip! I changed to your proof now and things got way shorter.  I interpreted the general statement as a filler of a square (with its corners being paths) instead of a filler of a cube, which I guess it should be (filling cubes can be so scary...). I don't know if this is super pretty, but I thought it made things easier to follow.

The only unfortunate thing is that I had to use a comp instead of an hcomp for the final statement because the general statement says something about refl*refl = refl*refl which does not reduce to refl = refl in cubical. But that should be fine.

I actually never had to use J. Maybe I proved a weaker version of the theorem without realising?

Reply:
Wow, the proof is really slick and cubical now!